src/System.Runtime/tests/System/Reflection/MethodBaseTests.cs

```diff
@@ -98,6 +113,13 @@ public static MethodBase MyFakeGenericMethod<T>()
             return m;
         }
 
+        public static MethodBase MyFakeGenericConstructedMethod<T>()
+        {
+            MethodBase m = MethodBase.GetCurrentMethod();
```

Comment:
I would avoid the use of GetCurrentMethod(). You can retrieve generic methods on types using any of the usual Type.GetMethod() overloads.